src/pages/playground/Playground.tsx

```diff
@@ -512,16 +549,23 @@ const Playground: React.FC<PlaygroundProps> = ({ workspaceLocation = 'playground
 
   const { handleEditorValueChange } = props;
 
-  // No point memoing this, it uses props.editorValue
-  const sessionButtons = (
-    <ControlBarSessionButtons
-      editorSessionId={props.editorSessionId}
-      // TODO: Hardcoded to make use of the first editor tab. Rewrite after editor tabs are added.
-      editorValue={props.editorTabs[0].value}
-      handleSetEditorSessionId={id => dispatch(setEditorSessionId(workspaceLocation, id))}
-      sharedbConnected={props.sharedbConnected}
-      key="session"
-    />
+  const getEditorValue = React.useCallback(
+    () => store.getState().workspaces.playground.editorTabs[0].value,
+    [store]
+  );
```

Comment:
We do this so that we only access the editor value when the session button is pressed, instead of constantly re-rendering the session button whenever the editor value changes. As a result, we need the `store` object using the `useStore` hook which does not listen to changes (as compared to the `useSelector` hook which does and will force re-renders every time the listened value updates).